internal/worker/server.go

```diff
@@ -131,11 +133,68 @@ func (s *Server) JobStatus(id uuid.UUID, result interface{}) (*JobStatus, []uuid
 		}
 	}
 
-	// For backwards compatibility: OSBuildJobResult didn't use to have a
-	// top-level `Success` flag. Override it here by looking into the job.
+	if r, ok := result.(*KojiInitJobResult); ok {
+		if r.JobResult.JobError == nil && r.KojiError != "" {
+			r.JobResult = JobResult{
+				JobError: clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, r.KojiError),
+			}
+		}
+	}
+
+	if r, ok := result.(*KojiFinalizeJobResult); ok {
+		if r.JobResult.JobError == nil && r.KojiError != "" {
+			r.JobResult = JobResult{
+				JobError: clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, r.KojiError),
+			}
+		}
+	}
+
+	if r, ok := result.(*OSBuildKojiJobResult); ok {
+		if r.JobResult.JobError == nil && r.KojiError != "" {
+			r.JobResult = JobResult{
+				JobError: clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, r.KojiError),
+			}
+		}
+	}
```

Comment:
This is more of a personal choice, but I find that case switching with types is more readable and easier to manager:
```go
	switch r := result.(type) {
	case *KojiInitJobResult:
		if r.JobResult.JobError == nil && r.KojiError != "" {
			r.JobResult = JobResult{
				JobError: clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, r.KojiError),
			}
		}
	case *KojiFinalizeJobResult:
		if r.JobResult.JobError == nil && r.KojiError != "" {
			r.JobResult = JobResult{
				JobError: clienterrors.WorkerClientError(clienterrors.ErrorOldResultCompatible, r.KojiError),
			}
		}
	case *OSBuildKojiJobResult:
		...
	}
```

You get the `r` variable cast to the compared type immediately, it avoids needing to check `ok` every time, it bundles all the type checks into one block, and it makes it easy to define a `default` case (if necessary).

Practically and functionally it's the same though, so I don't mind keeping it this way.  I just find this easier to manager personally.

Reply:
Yeah happy to make that change, I agree that it would make things more readable and easier to manage.